src/api/resolve-template.js

```diff
@@ -1,9 +1,11 @@
 const path = require('path');
+const semver = require('semver');
+const { getAppTemplateConfig } = require('../utils');
 
 function getTemplateNameByLibraryVersion(templateName, libraryVersion = '') {
   if (
     templateName === 'InstantSearch.js' &&
-    libraryVersion.substr(0, 2) === '2.'
+    semver.satisfies(libraryVersion, '>= 2.0.0 < 3.0.0')
```

Comment:
little bit doubting if we need this switching logic written manually, or if it can be something that can be done by querying all existing templates for that library, but 🆒  for now I think

Reply:
There might be a way to automate that, but let's not think too far ahead for now.